patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -152,36 +166,59 @@
          float f = this.getHealth();
          if (f > 0.0F) {
              this.setHealth(f + p_21116_);
-@@ -1118,6 +_,7 @@
- 
-     @Override
-     public boolean hurt(DamageSource p_21016_, float p_21017_) {
-+        if (!net.neoforged.neoforge.common.CommonHooks.onLivingAttack(this, p_21016_, p_21017_)) return false;
-         if (this.isInvulnerableTo(p_21016_)) {
+@@ -1127,23 +_,30 @@
+         } else if (p_21016_.is(DamageTypeTags.IS_FIRE) && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
              return false;
-         } else if (this.level().isClientSide) {
-@@ -1136,14 +_,17 @@
+         } else {
++            this.damageContainer.add(new net.neoforged.neoforge.common.damagesource.InternalDamageContainer(p_21016_, p_21017_));
++            if (!net.neoforged.neoforge.common.CommonHooks.onEntityIncomingDamage(this, this.damageContainer.getLast())) return false;
+             if (this.isSleeping() && !this.level().isClientSide) {
+                 this.stopSleeping();
+             }
+ 
+             this.noActionTime = 0;
++            p_21017_ = this.damageContainer.getLast().getNewDamage(); //Neo: enforce damage container as source of truth for damage amount
+             float f = p_21017_;
              boolean flag = false;
              float f1 = 0.0F;
-             if (p_21017_ > 0.0F && this.isDamageSourceBlocked(p_21016_)) {
+-            if (p_21017_ > 0.0F && this.isDamageSourceBlocked(p_21016_)) {
 -                this.hurtCurrentlyUsedShield(p_21017_);
 -                f1 = p_21017_;
 -                p_21017_ = 0.0F;
-+                net.neoforged.neoforge.event.entity.living.ShieldBlockEvent ev = net.neoforged.neoforge.common.CommonHooks.onShieldBlock(this, p_21016_, p_21017_);
-+                if(!ev.isCanceled()) {
-+                if(ev.shieldTakesDamage()) this.hurtCurrentlyUsedShield(p_21017_);
++            net.neoforged.neoforge.event.entity.living.LivingShieldBlockEvent ev;
++            if (p_21017_ > 0.0F && (ev = net.neoforged.neoforge.common.CommonHooks.onDamageBlock(this, this.damageContainer.getLast(), this.isDamageSourceBlocked(p_21016_))).getBlocked()) {
++                this.damageContainer.getLast().setBlockedDamage(ev);
```

Comment:
`Stack#peek`?

I'm not gonna do any more of these lol, just do a sweep if you decide to change the call